ospfd/ospfd.c

```diff
@@ -2070,13 +2073,14 @@ static int ospf_vrf_enable(struct vrf *vrf)
 				zlog_err("ospf_sock_init: could not raise privs, %s",
 					 safe_strerror(errno));
 			}
-			if (ospf_bind_vrfdevice(ospf, ospf->fd) < 0)
-				return 0;
+			ret = ospf_sock_init(ospf);
 			if (ospfd_privs.change(ZPRIVS_LOWER)) {
 				zlog_err("ospf_sock_init: could not lower privs, %s",
 					 safe_strerror(errno));
 			}
-
+			if (ret < 0 || ospf->fd <= 0)
+				return 0;
+			thread_add_read(master, ospf_read, ospf, ospf->fd, &ospf->t_read);
```

Comment:
May this test move right after call to ospf_sock_init() 5 lines before

Reply:
I did this so as to come back to lower priority.
this brings more consistency like that I think.
or the code should looke like following:


ospf_sock_init();
if (ret < 0 || fd < 0){
    change ZPRIVS_LOWER
   return 0;
}
change ZPRIVS_LOWER
thread_add_read

Reply:
Yes, Change sounds strange. Agree to not change and leave your original modification as is.